src/cmd/bundle.go

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package cmd contains the CLI commands for Zarf.
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/pkg/bundler"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var bundleCmd = &cobra.Command{
+	Use:     "bundle",
+	Aliases: []string{"b"},
+	Short:   lang.CmdBundleShort,
+}
+
+var bundleCreateCmd = &cobra.Command{
+	Use:     "create [DIRECTORY]",
+	Aliases: []string{"c"},
+	Args:    cobra.ExactArgs(1),
+	Short:   lang.CmdBundleCreateShort,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if !utils.IsDir(args[0]) {
+			return fmt.Errorf("first argument must be a valid path to a directory")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		bndlConfig.CreateOpts.SourceDirectory = args[0]
+
+		bndlConfig.CreateOpts.SetVariables = bundler.MergeVariables(v.GetStringMapString(V_PKG_CREATE_SET), bndlConfig.CreateOpts.SetVariables)
+
+		bndlClient := bundler.NewOrDie(&bndlConfig)
+		defer bndlClient.ClearPaths()
+
+		if err := bndlClient.Create(); err != nil {
+			message.Fatalf(err, "Failed to create bundle: %s", err.Error())
+		}
+	},
+}
+
+var bundleDeployCmd = &cobra.Command{
+	Use:     "deploy [BUNDLE]",
+	Aliases: []string{"d"},
+	Short:   lang.CmdBundleDeployShort,
+	Args:    cobra.ExactArgs(1),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if !utils.IsOCIURL(args[0]) && !bundler.IsValidTarballPath(args[0]) {
+			return fmt.Errorf("first argument must either be a valid OCI URL or a valid path to a bundle tarball")
+		}
+		return oci.ValidateReference(args[0])
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		bndlConfig.DeployOpts.Source = args[0]
+
+		bndlConfig.DeployOpts.SetVariables = bundler.MergeVariables(v.GetStringMapString(V_PKG_DEPLOY_SET), bndlConfig.DeployOpts.SetVariables)
+
+		bndlClient := bundler.NewOrDie(&bndlConfig)
+		defer bndlClient.ClearPaths()
+
+		if err := bndlClient.Deploy(); err != nil {
+			message.Fatalf(err, "Failed to deploy bundle: %s", err.Error())
+		}
+	},
+}
+
+var bundleInspectCmd = &cobra.Command{
+	Use:     "inspect [BUNDLE]",
+	Aliases: []string{"i"},
+	Short:   lang.CmdBundleInspectShort,
+	Args:    cobra.ExactArgs(1),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if !utils.IsOCIURL(args[0]) && !bundler.IsValidTarballPath(args[0]) {
+			return fmt.Errorf("first argument must either be a valid OCI URL or a valid path to a bundle tarball")
+		}
+		return oci.ValidateReference(args[0])
```

Comment:
This should probably be a helper function since it exists in a few spots. `PreRunE` can be set to a named function as well.